pkg/storage/legacysql/dualwrite/runtime.go

```diff
@@ -11,39 +11,40 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/registry/rest"
 
+	"github.com/grafana/grafana-app-sdk/logging"
+
 	grafanarest "github.com/grafana/grafana/pkg/apiserver/rest"
 )
 
-func (m *service) NewStorage(gr schema.GroupResource,
-	legacy grafanarest.Storage,
-	storage grafanarest.Storage,
-) (grafanarest.Storage, error) {
+func (m *service) NewStorage(gr schema.GroupResource, legacy grafanarest.Storage, unified grafanarest.Storage) (grafanarest.Storage, error) {
 	status, err := m.Status(context.Background(), gr)
 	if err != nil {
 		return nil, err
 	}
 
+	log := logging.DefaultLogger.With("gr", gr.String())
+
 	if m.enabled && status.Runtime {
 		// Dynamic storage behavior
 		return &runtimeDualWriter{
 			service:   m,
 			legacy:    legacy,
-			unified:   storage,
-			dualwrite: grafanarest.NewDualWriter(grafanarest.Mode3, legacy, storage, m.reg, gr.String()),
+			unified:   unified,
+			dualwrite: &dualWriter{legacy: legacy, unified: unified, log: log}, // not used for read
```

Comment:
Currently `mode1` is not really implemented with with the runtime version -- mode1 will write to both, but ignore  errors.

If we want to support that -- we will need an additional flag in the service

Reply:
I think you mean `mode0` from what I read in the code, right? 
https://github.com/grafana/grafana/blob/2cc6f39c5e6dbd62f1cd7a4defab3b1b189888fb/pkg/apiserver/rest/dualwriter.go#L84-L86

Which sould be fine for now. We can add it later if we need support for it.

Reply:
> I think you mean mode0 from what I read in the code, right?

we support mode0 with the runtime -- it is read/write legacy

we do not yet support mode1 with runtime, just mode2 where we report errors when writing to unified.

Honestly... really looking forward to stop talking about modes that we will never understand!